src/Symfony/Component/Config/Definition/VariableNode.php

```diff
@@ -81,6 +81,17 @@ protected function validateType($value)
      */
     protected function finalizeValue($value)
     {
+        if (!$this->allowEmptyValue && $this->isHandlingPlaceholder()) {
+            @trigger_error(sprintf('Using a placeholder value at path "%s" when empty values are not allowed by definition is deprecated since Symfony 4.2.', $this->getPath()), E_USER_DEPRECATED);
```

Comment:
What about

```
Setting path "%s" to an environment variable is deprecated since Symfony 4.2. Remove "cannotBeEmpty()" or include a non-empty string instead.
```

Reply:
Updated